library/src/main/java/org/xmtp/android/library/Contacts.kt

```diff
@@ -243,21 +298,42 @@ class ConsentList(
         return entry
     }
 
-    fun state(address: String): ConsentState {
+    suspend fun state(address: String): ConsentState {
+        client.v3Client?.let {
+            return ConsentState.fromFfiConsentState(
+                it.getConsentState(
+                    FfiConsentEntityType.ADDRESS,
+                    address
+                )
+            )
+        }
         val entry = entries[ConsentListEntry.address(address).key]
-
         return entry?.consentType ?: ConsentState.UNKNOWN
```

Comment:
Previously it was checking this local entries list which we could still just do since we write to both. But I like move towards checking the V3 state only since that should be the ultimate source of truth if you have V3.